platform/pwl.c

```diff
@@ -1393,10 +1417,12 @@ pwl_display_destroy (PwlDisplay *self)
         g_clear_pointer (&self->application_id, g_free);
         g_clear_pointer (&self->window_title, g_free);
 
-        g_clear_pointer (&self->xkb_data.state, xkb_state_unref);
-        g_clear_pointer (&self->xkb_data.compose_state, xkb_compose_state_unref);
-        g_clear_pointer (&self->xkb_data.keymap, xkb_keymap_unref);
-        g_clear_pointer (&self->xkb_data.context, xkb_context_unref);
+        wl_list_for_each_safe (item, tmp, &self->seats, link) {
```

Comment:
I don't like the same loop as above (line 1407) being repeated here. There is no reason why more than one loop should be needed.

Reply:
Code simplified and encapsulated in the `pwl_seat_destroy()`